File: api/src/main/java/me/devsnox/planetsystem/api/events/PlanetDeleteEvent.java
package me.devsnox.planetsystem.api.events;

import me.devsnox.planetsystem.api.planet.Planet;
import org.bukkit.event.Event;
import org.bukkit.event.HandlerList;

public class PlanetDeleteEvent extends Event {

    private static final HandlerList handlers = new HandlerList();
    private final Planet planet;

    public PlanetDeleteEvent(final Planet planet) {
        this.planet = planet;
    }

    @Override
    public HandlerList getHandlers() {
        return handlers;
    }

    public static HandlerList getHandlerList() {
        return handlers;
    }

    public Planet getPlanet() {
        return planet;
    }
}

File: api/src/main/kotlin/de/astride/planetsystem/api/events/PlanetDeleteEvent.kt
package de.astride.planetsystem.api.events

import de.astride.planetsystem.api.planet.LoadedPlanet
import org.bukkit.event.Event
import org.bukkit.event.HandlerList

class PlanetDeleteEvent(val planet: LoadedPlanet) : Event() {

	override fun getHandlers(): HandlerList = handlerList

	companion object {

		@JvmStatic
		val handlerList = HandlerList()

	}

}
